starfish/image/_single.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import collections
+from typing import Tuple
+
+from slicedimage import ImageFormat
+from slicedimage.io import resolve_url
+
+from ._base import ImageBase
+
+
+class SingleImage(ImageBase):
+    def __init__(self, data):
+        self._data = data
+
+    @classmethod
+    def from_url(cls, relativeurl: str, baseurl: str):
+        backend, name, baseurl = resolve_url(relativeurl, baseurl)
+        extension = os.path.splitext(relativeurl)[1].lstrip(".")
+        format = ImageFormat.find_by_extension(extension)
+        with backend.read_file_handle(name) as fh:
+            return SingleImage(format.reader_func(fh))
+
+    @property
+    def numpy_array(self):
+        data = self._data.view()
+        data.setflags(write=False)
+        return data
+
+    @numpy_array.setter
+    def numpy_array(self, data):
+        self._data = data.view()
+        data.setflags(write=False)
+
+    @property
+    def raw_shape(self) -> Tuple[int]:
+        return self._data.shape
+
+    @property
+    def shape(self) -> collections.OrderedDict:
```

Comment:
Does this return an empty dict because there are no `h, c, z` indices for a single image? If so I'd add a docstring that explains why this happens -- I could see it confusing users.